micro-rdk/src/esp32/conn/network.rs

```diff
@@ -202,25 +202,54 @@ impl Esp32WifiNetwork {
 
         let sl_stack = esp32_get_system_event_loop()?;
 
-        let subscription = sl_stack.subscribe::<WifiEvent, _>(move |event: WifiEvent| {
-            if matches!(event, WifiEvent::StaDisconnected(_)) {
-                if let Ok(wifi) = esp32_get_wifi() {
-                    if let Some(mut wifi_guard) = wifi.try_lock() {
-                        let wifi_mut = wifi_guard.wifi_mut();
-                        if let Err(err) = wifi_mut.connect() {
-                            let ssid = wifi_mut
-                                .get_configuration()
-                                .map_or("<no_ssid>".to_owned(), |c| {
-                                    c.as_client_conf_ref().unwrap().ssid.to_string()
-                                });
-                            log::error!("could not connect to WiFi {} cause : {:?}", ssid, err);
+        let subscription =
+            sl_stack.subscribe::<WifiEvent, _>(move |event: WifiEvent| match event {
+                WifiEvent::StaDisconnected(disconnected) => {
+                    let ssid: String = CString::new(disconnected.ssid())
```

Comment:
I believe there is no restriction against having null characters in an SSID, so this would truncate. What other options are there for logging a `&[u8]` that don't require first converting into a CString?